dev/com.ibm.ws.session.cache/src/com/ibm/ws/session/store/cache/CacheHashMap.java

```diff
@@ -139,103 +139,109 @@
 
-                sessionAttributeCache = cacheStoreService.cacheManager.createCache(attrCacheName, config);
-            } catch (CacheException x) {
-                create = false;
-                if (trace && tc.isDebugEnabled()) {
-                    tcReturn(cacheStoreService.tcCacheManager, "createCache", x);
-                    tcInvoke(cacheStoreService.tcCacheManager, "getCache", attrCacheName, "String", "byte[]");
+                MutableConfiguration<String, byte[]> config = new MutableConfiguration<String, byte[]>()
+                                .setTypes(String.class, byte[].class)
+                                .setExpiryPolicyFactory(EternalExpiryPolicy.factoryOf());
+                if (cacheStoreService.supportsStoreByReference)
+                    config = config.setStoreByValue(false);
+                try {
+                    if (trace && tc.isDebugEnabled())
+                        tcInvoke(cacheStoreService.tcCacheManager, "createCache", attrCacheName, config);
+
+                    sessionAttributeCache = cacheStoreService.cacheManager.createCache(attrCacheName, config);
+                } catch (CacheException x) {
+                    create = false;
+                    if (trace && tc.isDebugEnabled()) {
+                        tcReturn(cacheStoreService.tcCacheManager, "createCache", x);
+                        tcInvoke(cacheStoreService.tcCacheManager, "getCache", attrCacheName, "String", "byte[]");
+                    }
+                    sessionAttributeCache = cacheStoreService.cacheManager.getCache(attrCacheName, String.class, byte[].class);
+                    if (sessionAttributeCache == null)
+                        throw x;
                 }
-                sessionAttributeCache = cacheStoreService.cacheManager.getCache(attrCacheName, String.class, byte[].class);
-                if (sessionAttributeCache == null)
-                    throw x;
             }
-        }
 
-        tcSessionAttrCache = "AttrCache" + Integer.toHexString(System.identityHashCode(sessionAttributeCache));
-        if (trace && tc.isDebugEnabled())
-            tcReturn(cacheStoreService.tcCacheManager, create ? "createCache" : "getCache", tcSessionAttrCache, sessionAttributeCache);
+            tcSessionAttrCache = "AttrCache" + Integer.toHexString(System.identityHashCode(sessionAttributeCache));
+            if (trace && tc.isDebugEnabled())
+                tcReturn(cacheStoreService.tcCacheManager, create ? "createCache" : "getCache", tcSessionAttrCache, sessionAttributeCache);
 
-        cacheStoreService.configureMonitoring(attrCacheName);
+            cacheStoreService.configureMonitoring(attrCacheName);
+        } catch(Exception ex) {
+            FFDCFilter.processException(ex, "com.ibm.ws.session.store.cache.CacheHashMap.cacheInit", "241");
+            Tr.error(tc, "ERROR_SESSION_INIT", ex);
+            throw new RuntimeException("Internal Server Error");
```

Comment:
Exception messages should always be translated if surfacing to the user, which I believe is the case because it was the whole point of these changes